### ISSUE TITLE
Implement menus on X11

### ISSUE THREAD

**Issue by emoon** [emoon/ProDBG#163]
Right now menus uses the native menus on Mac and on Windows which I think is desirable. On Linux/BSD/X11 there are no native menus and they depend on which window manager that is being used. This makes it hard to target 1 API. My suggestion is the following

1. On Linux use the dear imgui menus instead 
![test_window_05_menus](https://cloud.githubusercontent.com/assets/350822/14910295/5ed33f02-0dec-11e6-844e-46cf2e73c9bf.png) 
We already use this API for all other stuff so should be a good fit.
Problem is that all menu setup is inside minifb. It's not hard to get access to the data but there needs to be a some custom code written to use the data for dear imgui.

2. Depend on something like GTK or such.

I would rather go with option 1 though.

**Comment by kondrak** [comment]
Voting for 1. It may be slightly more cumbersome to implement but it'll work out of the box on every platform and it's a lesser dependency than GTK.

**Comment by emoon** [comment]
Agreed. esp the fact that is cross-platform is a great thing if the case of porting it to other platforms come up that would help as well.

**Comment by emoon** [comment]
So to implement this on Linux:

https://github.com/emoon/ProDBG/blob/master/src/prodbg/main/src/menu.rs has all the menu data. This is being sent to minifb which sets up the menus and such. Later one in the code one can call

```Rust
self.win.is_menu_pressed().map(|menu_id| {
   println!("Some one pressed menu {}", menu_id);
});

The way I envision this to work on Linux is to start with a separate function (in Window) that does something similar so it would be something like

```Rust
    #[cfg(any(target_os="macosx",
              target_os="windows",
    fn is_menu_pressed(&self) -> Option<usize) {
        self.win.is_menu_pressed()
    }
    #[cfg(any(target_os="linux",
              target_os="freebsd",
              target_os="dragonfly",
              target_os="netbsd",
              target_os="openbsd"))]
        fn is_menu_pressed(&self) -> Option<usize> {
            let ui = Imgui::get_ui();
            // implement the menus using imgui and the ui instance here
        }
```

**Comment by emoon** [comment]
Hold of a bit with implementing this as I'm re-doing the API for menus in minfb. Now Mac and Windows does calls directly to the native APIs and doesn't take a full structure of data in as before. The way I will handle this on Linux (where menus aren't supported) is that the calls will instead build a data structure that can be fetched back later for Linux.

That way ProDBG (and other apps using minifb) can setup the menus the way the want under Linux.

**Comment by emoon** [comment]
As of https://github.com/emoon/ProDBG/commit/a3427bcc188623f1e388da73faa5061d18c86760 ProDBG now uses minifb 0.7.0. Menu API has been fully rewritten in minifb now so for Linux/Unix it's possible to use http://prodbg.com/minifb/minifb/struct.Window.html#method.get_unix_menus which will provide back Menu data that can be used with Imgui

**Comment by emoon** [comment]
Made some preparation

https://github.com/emoon/ProDBG/blob/master/src/prodbg/main/src/windows.rs#L320

Code should work to just add in there

**Comment by SlNPacifist** [comment]
I guess this should be closed via #275

**Comment by emoon** [comment]
Yeah. 

Closed by https://github.com/emoon/ProDBG/commit/f501ea883fdbda75cc5e636d6894e9d6751264e5
